Add support for the FT4232h

davidlattimore opened the pull request:
Note, this is a breaking change, since it adds type parameters to structs such as I2c. I don't have a 4232h (although I'll get getting one in about a month). I tested this change with an ft2232 making an i2c request

davidlattimore committed:
Add support for the FT4232h

Note, this is a breaking change, since it adds type parameters to
structs such as I2c

newAM reviewed (changes requested):
Looks good to me!

There are some unresolved links now (I need to add a documentation check to CI), can you update these?

Reproducible with `cargo doc`:

```
warning: unresolved link to `self::Ft232hHal::ad0`
 --> src/gpio.rs:9:25
  |
9 | /// [`Ft232hHal::ad0`]: crate::Ft232hHal::ad0
  |                         ^^^^^^^^^^^^^^^^^^^^^ the type alias `Ft232hHal` has no associated item named `ad0`
  |
  = note: `#[warn(broken_intra_doc_links)]` on by default

warning: unresolved link to `self::Ft232hHal::ad7`
  --> src/gpio.rs:10:25
   |
10 | /// [`Ft232hHal::ad7`]: crate::Ft232hHal::ad7
   |                         ^^^^^^^^^^^^^^^^^^^^^ the type alias `Ft232hHal` has no associated item named `ad7`

warning: unresolved link to `self::Ft232hHal::i2c`
  --> src/i2c.rs:17:25
   |
17 | /// [`Ft232hHal::i2c`]: crate::Ft232hHal::i2c
   |                         ^^^^^^^^^^^^^^^^^^^^^ the type alias `Ft232hHal` has no associated item named `i2c`

warning: unresolved link to `self::Ft232hHal::spi`
  --> src/spi.rs:10:25
   |
10 | /// [`Ft232hHal::spi`]: crate::Ft232hHal::spi
   |                         ^^^^^^^^^^^^^^^^^^^^^ the type alias `Ft232hHal` has no associated item named `spi`

warning: 4 warnings emitted
```

davidlattimore commented:
Thanks. I've fixed the links. I also changed the `pub use` on `Ft232h` and `Ft4232h` back to just a regular `use`, since I realized that you were already exporting the whole `libftd2xx` library. I also updated the README.md to add FT4232H to the list of supported devices.

I wasn't sure whether you prefer to separate commits during review or amended commits. I've amended my commit for now, but can repush as separate commits if you prefer

newAM reviewed (approved):
Excellent, thanks!

I will probably get around to a release sometime next week - I still need to sort out the linking updates I made in `libftd2xx` for this crate.
